src/adapters/encounter-diagnosis-adapter.ts

```diff
@@ -49,47 +68,46 @@ export const EncounterDiagnosisAdapter: FormFieldValueAdapter = {
   },
 };
 
-const constructNewDiagnosis = (value: any, field: FormField, patientUuid: string, uuid?: string) => {
+const constructNewDiagnosis = (value: any, field: FormField, patientUuid: string) => {
   if (!value) {
     return null;
   }
-  const diagnosis: DiagnosisPayload = {
+  return {
     patient: patientUuid,
     condition: null,
     diagnosis: {
       coded: value,
     },
-    certainty: field.questionOptions?.diagnosis?.isConfirmed ? 'CONFIRMED' : 'PROVISIONAL',
+    certainty: isTrue(field.questionOptions?.diagnosis?.isConfirmed) ? 'CONFIRMED' : 'PROVISIONAL',
     rank: field.questionOptions.diagnosis?.rank ?? 1, // rank 1 denotes a diagnosis is primary, else secondary
     formFieldPath: `rfe-forms-${field.id}`,
     formFieldNamespace: 'rfe-forms',
   };
-
-  if (uuid && uuid.trim() !== '') {
-    diagnosis.uuid = uuid;
-  }
-
-  return diagnosis;
 };
 
-function editDiagnosis(newEncounterDiagnosis: any, field: FormField, patientUuid: string) {
-  if (newEncounterDiagnosis === field.meta.previousValue?.diagnosis?.coded?.uuid && !field.meta.repeat?.wasDeleted) {
-    clearSubmission(field);
-    return null;
-  }
+function editDiagnosis(
+  newEncounterDiagnosis: any,
+  field: FormField,
+  previousDiagnosis: OpenmrsResource,
+  patientUuid: string,
+) {
+  return {
+    patient: patientUuid,
+    condition: null,
+    diagnosis: {
+      coded: newEncounterDiagnosis,
+    },
+    certainty: isTrue(field.questionOptions?.diagnosis?.isConfirmed) ? 'CONFIRMED' : 'PROVISIONAL',
+    rank: field.questionOptions.diagnosis?.rank ?? 1, // rank 1 denotes a diagnosis is primary, else secondary
+    formFieldPath: `rfe-forms-${field.id}`,
+    formFieldNamespace: 'rfe-forms',
+    uuid: previousDiagnosis.uuid,
+  };
+}
 
-  //the field has been deleted
-  if (field.meta.repeat?.wasDeleted) {
-    const voided = {
-      uuid: field.meta.previousValue?.uuid,
-      voided: true,
-    };
-    gracefullySetSubmission(field, constructNewDiagnosis(newEncounterDiagnosis, field, null), voided);
-    return field.meta.submission.newValue || null;
-  } else {
-    const oldDiagnosis = field.meta.initialValue?.omrsObject as OpenmrsResource;
-    const newValue = constructNewDiagnosis(newEncounterDiagnosis, field, patientUuid, oldDiagnosis.uuid);
-    gracefullySetSubmission(field, newValue, null);
-    return newValue;
+export function hasPreviousDiagnosisValueChanged(previousDiagnosis: OpenmrsResource, newValue: any) {
```

Comment:
Is the value for `newValue` a string (some concept UUID)? If so, can you update the typing?